### PR TITLE
Fix migration for orphaned Puppetclass associations

### PR DESCRIPTION
**Pull request by ezr-ondrej**

### CODE REVIEW

**Review by kamils-iRonin — APPROVED** [review]
I think the code looks good, @laugmanuel would you be able to test it?

File: db/migrate/20200803113903_migrate_host_type_in_host_config_groups.foreman_puppet.rb [hunk 1]
```diff
@@ -3,16 +3,27 @@ class FakeHostConfigGroup < ::ApplicationRecord
     self.table_name = 'host_config_groups'
```
[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply db/migrate/20200803113903_migrate_host_type_in_host_config_groups.foreman_puppet.rb:3]: Looks much better :)

**Comment by laugmanuel** [comment]
How does one manage to get orphan puppetclass associations? I'm trying to test this but not sure how to do that properly.

**Review by laugmanuel — APPROVED** [review]

**Comment by ezr-ondrej** [comment]
CP 1.0: 69b3e1f

**Pull request merged**